doc/hooks/patch-activation-hooks.section.md

```diff
@@ -0,0 +1,51 @@
+
+# `patchActivationBash`, `patchActivationCsh`, `patchActivationFish`, `patchActivationPosix` {#sec-patchActivationHooks}
+
+These hooks provide utilities to patch
+shell scripts meant to be sourced (e.g. the "activate" scripts).
+
+It inserts two sections of codes into the beginning and the end of the script:
+the former inserts a `PATH` prefix, while the latter clears it up.
+This way, the original commands will have access to the executables needed,
+but how `PATH` variable is changed by the script remains the same after patching.
+
+A stack is used to record the inserted `PATH` prefixes when patched multiple times,
+so that a patched activation script can correctly source another patched script.
+
+The utilities are called as
+
+```sh
+patchActivation<Shellname> <file> <PATH-prefix>
+```
+
+Example usage:
+
+```nix
+{ lib, stdenv, patchActivationBash}:
+stdenv.mkDerivation {
+
+  # ...
+
+  nativeBuildInputs = [
+    patchActivationBash
+  ];
+
+  postFixup = ''
+    patchActivationBash $out/bin/activate-foo.zsh ${lib.makeBinPath [ coreutils man which ]}
```

Comment:
This needs an example of `activate-foo` to make clear why we call it that way. Something like

> Given a build hook `foo` that depends on `sed`, `man` and `which` (it seems we may need a more sensible example), patch the hook to have its dependencies available without interfering with the rest of the `setup` environment.

Reply:
Thanks a lot for the example context.

The hook is to patch the scripts meant to be sourced by the software users, not the implementation of another setup hook. A setup hook can get all it needs by the `substitutions` attribute and the `@foo@` syntax, while the target scripts are provided by the upstream that might not be easily patched through string-find-and-replacing.